bitcoin/src/blockdata/witness.rs

```diff
@@ -95,16 +108,37 @@ impl Decodable for Witness {
                 cursor += element_size;
             }
             content.truncate(cursor);
+            // Index space is now at the end of the Vec
+            content.rotate_left(witness_index_space);
             Ok(Witness {
                 content,
                 witness_elements,
-                last,
-                second_to_last,
+                indices_start: cursor - witness_index_space,
             })
         }
     }
 }
 
+
+/// Safety Requirements: value must always fit within u32
```

Comment:
In 4226d602057529885cf95f935370f6e5bf68bfd2:

This is not a safety requirement, but a correctness requirement.